lib/allflex/product_mapper/product_builder.rb

```diff
@@ -27,7 +27,7 @@ def build(ordered_product, product_defn)
                                       components:         components.value_or,
                                       manufacturing_definition: product_defn
                                     )
-        product.id.success? ? Success(product) : Failure(product)
+        product.id.success? ? Success(product) : Failure(product_id_building: [{ code: 'Allflex Product Creation Error', errors: { message: 'Error in build method in Product Builder' }, context: { product: product } }])
```

Comment:
Where is the test for this change?

```suggestion
        product.id.success? ? Success(product) : Failure(product_id_building: [{ code: 'Allflex Product Creation Error', errors: [{ message: 'Error in build method in Product Builder' }], context: { product: product } }])
```

Reply:
Haven't managed to be able to pull a test for this one sorry. Couldn't replicate it as it gets caught earlier and never goes to the Failure here from the specs we have.
Leave for now i think.